Add Auth

Taps73r opened the pull request.

Taps73r committed:
Add Auth controllers, services. Fix User model, controllers, services

Taps73r committed:
docs update

DmytroProg commented on src/Service/AuthService.ts:
```diff
@@ -0,0 +1,46 @@
+            const token = this.generateToken(user);
+            return { role: user.role, token };
+        }
+        return null;
```
throw exception instead of null

DmytroProg commented on src/Model/User.ts:
```diff
@@ -2,19 +2,27 @@ import { Entity, PrimaryGeneratedColumn, Column } from "typeorm";
     username: string;
 
+    @Column()
+    role: "User" | "Admin";
```
save numeric type

Taps73r committed:
Implement changes based on PR comments

- Moved error handling to a separate ErrorHandler class
- Created a utils directory to store the ErrorHandler
- Refactored AuthController to use the ErrorHandler for consistent error handling
- Ensured proper error logging in AuthService
- Changed role type from string to number in AuthService
- Fixed AuthService login method to ensure it doesn't return 0 for role

Taps73r committed:
Code refactoring

Taps73r committed:
Add admin registration and role-based authorization
- Added UserRole enum for user and admin roles
- Updated User model to include role field
- Enhanced AuthService with registerAdmin method for admin creation
- Updated AuthController to include registerAdmin endpoint
- Added middleware to authenticate and authorize admin users
- Protected /register-admin route with role-based authorization
- Implemented JWT payload to include user roles
- Added error handling for duplicate users during registration
